test/common/armoire.js

```diff
@@ -0,0 +1,182 @@
+        armoireKey = gearObject.key;
+        fullArmoire[armoireKey] = true;
+      });
+    });
+
+    beforeEach(function() {
+      user.achievements.ultimateGearSets = { rogue: true };
+      user.flags.armoireOpened = true;
+      user.stats.exp = 0;
+      user.items.food = {};
+    });
+
+    it('does not open if user does not have enough gold', function(done) {
+      user.fns.predictableRandom.returns(YIELD_EQUIPMENT);
+      user.stats.gp = 50;
+
+      user.ops.buy({params: {key: 'armoire'}}, function(response) {
+        expect(response.message).to.eql('Not Enough Gold');
+        expect(user.items.gear.owned).to.eql({'weapon_warrior_0': true});
+        expect(user.items.food).to.be.empty;
+        expect(user.stats.exp).to.eql(0);
+        done();
+      });
+    });
+
+    it('does not open without Ultimate Gear achievement',function(done) {
+      user.fns.predictableRandom.returns(YIELD_EQUIPMENT);
+      user.achievements.ultimateGearSets = {'healer':false,'wizard':false,'rogue':false,'warrior':false};
+
+      user.ops.buy({params: {key: 'armoire'}}, function(response) {
+        expect(response.message).to.eql("You can't buy this item");
+        expect(user.items.gear.owned).to.eql({'weapon_warrior_0': true});
+        expect(user.items.food).to.be.empty;
+        expect(user.stats.exp).to.eql(0);
+        done();
+      });
+    });
+
+    it('always drops equipment the first time', function() {
+      delete user.flags.armoireOpened;
+      user.fns.predictableRandom.returns(YIELD_EXP);
+
+      user.ops.buy({params: {key: 'armoire'}});
+
+      expect(user.items.gear.owned).to.eql({
+        'weapon_warrior_0': true,
+        'shield_armoire_gladiatorShield': true
+      });
+
+      var armoireCount = shared.count.remainingGearInSet(user.items.gear.owned, 'armoire');
+      expect(armoireCount).to.eql (_.size(fullArmoire) - 1)
+      expect(user.items.food).to.be.empty;
+      expect(user.stats.exp).to.eql(0);
+      expect(user.stats.gp).to.eql(100);
+    });
+
+    it('gives Experience', function() {
+      user.fns.predictableRandom.returns(YIELD_EXP);
+
+      user.ops.buy({params: {key: 'armoire'}})
+
+      expect(user.items.gear.owned).to.eql({'weapon_warrior_0': true});
+      expect(user.items.food).to.be.empty;
+      expect(user.stats.exp).to.eql(46);
+      expect(user.stats.gp).to.eql(100);
+    });
+
+    it('gives food', function() {
+      user.fns.predictableRandom.returns(YIELD_FOOD);
+
+      user.ops.buy({params: {key: 'armoire'}})
+
+      expect(user.items.gear.owned).to.eql({'weapon_warrior_0': true});
+      expect(user.items.food).to.eql({'Honey': 1});
+      expect(user.stats.exp).to.eql(0);
+      expect(user.stats.gp).to.eql(100);
+    });
+
+    it('gives more equipment', function() {
+      user.fns.predictableRandom.returns(YIELD_EQUIPMENT);
+      user.items.gear.owned = {
+        weapon_warrior_0: true,
+        head_armoire_hornedIronHelm: true
```

Comment:
This is another piece I've had to edit manually, if we can figure out how to correctly predict what the new equipment yield should be when available items change.